File: frontend/src/features/workItems/model/workItem.ts
export type WorkItemStatus = 'new' | 'in_progress' | 'blocked' | 'done';

export type WorkItemPriority = 'low' | 'medium' | 'high' | 'critical';

export type WorkItem = {
  id: string;
  title: string;
  status: WorkItemStatus;
  priority: WorkItemPriority;
  assignee: string | null;
  tags: string[];
  revision: number;
  updatedAt: string;
};

export const workItemStatusLabels: Record<WorkItemStatus, string> = {
  new: 'New',
  in_progress: 'In progress',
  blocked: 'Blocked',
  done: 'Done',
};

export const workItemPriorityLabels: Record<WorkItemPriority, string> = {
  low: 'Low',
  medium: 'Medium',
  high: 'High',
  critical: 'Critical',
};

File: backend/src/main/java/com/ewsreference/app/workitem/domain/WorkItem.java
package com.ewsreference.app.workitem.domain;

import java.time.Instant;
import java.util.List;

public record WorkItem(
        String id,
        String title,
        WorkItemStatus status,
        WorkItemPriority priority,
        String assignee,
        List<String> tags,
        long revision,
        Instant updatedAt
) {
    public WorkItem {
        tags = List.copyOf(tags);
    }
}
